spinnaker/resource_application.go

```diff
@@ -143,66 +148,46 @@ func resourceSpinnakerApplicationRead(d *schema.ResourceData, meta interface{})
 	if v := app.Attributes.Permissions; v != nil {
 		tfPermissions, err := buildTerraformPermissions(v)
 		if err != nil {
-			return err
+			return diag.FromErr(err)
 		}
 
 		d.Set("permissions", tfPermissions)
 	}
 
-	return nil
+	return diags
 }
 
-func resourceSpinnakerApplicationUpdate(d *schema.ResourceData, meta interface{}) error {
+func resourceSpinnakerApplicationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	clientConfig := meta.(gateConfig)
 	client := clientConfig.client
 	task, err := api.NewCreateApplicationTask(d)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	if err := api.CreateApplication(client, task); err != nil {
-		return err
+		return diag.FromErr(err)
 	}
-	return resourceSpinnakerApplicationRead(d, meta)
+	return resourceSpinnakerApplicationRead(ctx, d, meta)
 }
 
-func resourceSpinnakerApplicationDelete(d *schema.ResourceData, meta interface{}) error {
+func resourceSpinnakerApplicationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	clientConfig := meta.(gateConfig)
+	var diags diag.Diagnostics
 	client := clientConfig.client
 	appName := api.GetApplicationName(d)
 
 	if err := api.DeleteApplication(client, appName); err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	d.SetId("")
-	return nil
-}
-
-func resourceSpinnakerApplicationExists(d *schema.ResourceData, meta interface{}) (bool, error) {
```

Comment:
[note]
[Deprecation of helper/schema.ExistsFunc](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#deprecation-of-helper-schema-existsfunc)

> The helper/schema.ExistsFunc (usually set as the Exists property on helper/schema.Resource) is now deprecated.